src/providers/FileSystemProvider/FileSystemProvider.ts

```diff
@@ -284,19 +284,14 @@ export class FileSystemProvider implements vscode.FileSystemProvider {
   }
 
   public async readDirectory(uri: vscode.Uri): Promise<[string, vscode.FileType][]> {
-    uri = redirectDotvscodeRoot(uri);
+    if (uri.path.includes(".vscode/")) {
```

Comment:
This prevents VS Code from sending a StudioOpenDialog query request on the "_vscode" folder, which won't return anything. I tested that VS Code still uses the server-side settings.json file if it exists with this check added.